vultr/resource_vultr_iso_private.go

```diff
@@ -107,7 +109,71 @@ func resourceVultrIsoDelete(ctx context.Context, d *schema.ResourceData, meta in
 	log.Printf("[INFO] Deleting iso : %s", d.Id())
 
 	if err := client.ISO.Delete(ctx, d.Id()); err != nil {
-		return diag.Errorf("error destroying ISO %s : %v", d.Id(), err)
+		// decode the error
+		var attachedErr struct {
+			Error  string `json:"error"`
+			Status int    `json:"status"`
+		}
+
+		errR := strings.NewReader(err.Error())
+		jsonErr := json.NewDecoder(errR).Decode(&attachedErr)
```

Comment:
Could change this to json.Unmarshal 

```go
var attachedErr struct {
		Error  string `json:"error"`
		Status int    `json:"status"`
	}

	if err := json.Unmarshal([]byte(e.Error()), &attachedErr); err != nil {
		fmt.Println("error")
	}
```

Reply:
Yeah, that looks better than what I had.

Please see https://github.com/vultr/terraform-provider-vultr/pull/131/commits/0abc7c92d50e36870cd64e2bdf2461f15b58ef69